Makefile

```diff
@@ -124,11 +124,29 @@ test-containerized: run-etcd run-k8s-apiserver build-containerized dist/host-loc
 	-e PLUGIN=calico \
 	-e CNI_SPEC_VERSION=$(CNI_SPEC_VERSION) \
 	-v $(CURDIR):/go/src/github.com/projectcalico/cni-plugin:rw \
+	-v $(CURDIR)/dist:/opt/cni/bin \
 	$(CALICO_BUILD) sh -c '\
 			cd  /go/src/github.com/projectcalico/cni-plugin && \
 			ginkgo'
 	make stop-etcd
 
+.PHONY: k8s-install/scripts/install_cni.test
+## We pre-build the test binary so that we can run it outside a container and allow it
+## to interact with docker.
+k8s-install/scripts/install_cni.test: vendor
+	-mkdir -p .go-pkg-cache
+	docker run --rm \
+	-e LOCAL_USER_ID=$(LOCAL_USER_ID) \
+	-v $(CURDIR):/go/src/github.com/projectcalico/cni-plugin:rw \
+	-v $(CURDIR)/.go-pkg-cache:/go/pkg/:rw \
+		$(CALICO_BUILD) sh -c '\
+			cd /go/src/github.com/projectcalico/cni-plugin && \
+			go test ./k8s-install/scripts -c --tags install_cni_test -o ./k8s-install/scripts/install_cni.test'
+
+.PHONY: test-install-cni
+## Test the install-cni.sh script
```

Comment:
These ones are fine to have as `##` since it's a "user-facing" make target we'd want to show up in `make help`